### DISCUSSION TITLE
Improved discoverablity within the prompt mode.

### DISCUSSION THREAD

**Discussion by slembcke** [helix-editor/helix#9663] [Ideas]
So I came across Helix recently, and it seems pretty nice. I have a very neutral opinion on Vi, and the 4 commands I can remember make it useful in a pinch, but not something I'd use everyday. Helix on the other hand I was able to get working with clangd in minutes, and I felt halfway dangerous even without really knowing the keys or commands... Which then became the problem, discovering keys or commands meant repeatedly re-opening the website and searching. The command mode _almost_ fills in this gap, but I feel there is a missed opportunity regarding the autocomplete navigation and such.

For example, I could type `:op` and see there are a bunch of open commands, but I couldn't figure out how to open the file picker seen in the video on the website. I probably would have discovered the space mode sooner if I'd read the docs front to back, but I didn't notice it until a week later on my 15th trip back to the website. >_> I had a similar experience trying to figure out how to enable word wrapping, or just finding keys to do certain actions. It seems like there should be a keymap reference within Helix, but it seems not?

As a concrete list of examples, it would be great if:
* the file picker showed up in the suggestion list when I type `:open`, and told me it's bound to <space> f
* tying `:open` would say something like `usage: open filename`, instead of `invalid argument count`
* a ready to go `set soft-wrap.enable true` suggestion showed up if I typed `:wrap`
* other keys could be discovered this way too. ex: `:cap` could suggest the capitalize command and tell me it's bound to `~`
* command suggestions could be clickable, pressing tab 10 times is a little frustrating

I suspect with some changes like this to make the features more discoverable, I could have felt productive enough in Helix after a couple days. Otherwise, as a modal text novice I still feel slow a week later, and still need to leave Helix and open the website every time I have a question.

**Comment by the-mikedavis** [comment]
The command mode (`:`) popup only covers typable commands - commands that you can execute from that prompt. For all commands try the command picker (`<space>?`). Also see the tutor (`hx --tutor` or the `:tutor` command) if you haven't already

**Reply by slembcke** [reply]
I did know about `--tutor`, but I kind of found it by accident. I did _not_ know about the command palette, and now that I know it exists the documentation still makes it [basically invisible](https://docs.helix-editor.com/keymap.html?highlight=command%20palette#space-mode). 

Bear with me, an enthusiastic novice user, for a moment. I think the only thing Helix is missing to help onboard new users is a little text at launch. Instead of launching to an empty buffer, default to a short "splash" text. Tell the user about `:tutor`, how to quit (!!), tell them about space mode, and maybe most important of all tell them about the command palette! That's basically what VSCode does. It front and center tells you how to discover everything it can do without having to leave the editor. It seems like Helix has already done most of the hard work, it just needs some well placed words to tell novice users about them.

Experienced users probably like that it launches straight into a blank buffer, so give them a config parameter to disable the splash text. You could even document the `:config-open` command and the variable to set right there. : D
